### ISSUE TITLE
receiving song titles in wrong order

### ISSUE THREAD

**Issue by bluespark23** [dlrudie/Snip#410]
since update to Spotify 1.1.26.501.gbe11e53b  I now get song title at the end of a song instead of the beginning. so now I get a pop up that tells me what just finished.
I am using Snip 7.0.5
has something changed? tried reinstalling both to no avail - getting same behavior.
I have been a fan of Snip for many years. it's usually Spotify that mess things up !

**Comment by murko69** [comment]
I have 1.1.26.501 (Standalone) and Snip 7.05
No issues on the updating here. All text files updates on track change as usual (with info on the track that just started) and I just enabled the option about the notification box just to check - also works fine here.

**Comment by bluespark23** [comment]
Thanks for checking, murko69.  Will continue to investigate.

**Comment by bluespark23** [comment]
Still getting the same after reinstalling snip and spotify.  I found the first few songs title correctly , then it misses one and puts  the next song up at the end. this seems to happen with notifications on or off. It  is very strange, never had this before.

**Comment by dlrudie** [comment]
Can you link me to the songs it does this on so I can test?

**Comment by bluespark23** [comment]
Thanks David, I will pass a list - I  wondered if it was something to do with the actual titles but it didn't seem to be related.

**Comment by dlrudie** [comment]
It could be. Won’t know until I can test it. It’s been awhile since I thoroughly tested Snip out. I used it heavily about a month or so ago and it still worked fine, but I was also playing the same songs I always play. I’ve ran into issues with the titles in the past so you never know. It may also not be the title at all and maybe something else in the JSON data.

If you can get me the list in the next few hours I can try it out tonight. Otherwise it will be tomorrow after I’m off work.

Thanks for the support!

**Comment by bluespark23** [comment]
My Spotify updated to  1.1.27.472.gf6574b9 today,  and the problem seems to have gone away.
Just tried a list with 25 songs... all OK now. Thanks for your consideration.

**Comment by murko69** [comment]
@dlrudie 
Actually...... This started to happen for me aswell in the last week.
Firstly I just ignored it and hit pause/play and didn't think of it again, but the next time I noticed it I thought I would let Spotify just play and see what happens and very true - just after the song is finished and the next track starts the text file is updated with the previous track and not the present one.

I'm playing pretty much the same tracks that I did when it didn't happen at all,
Also I've tested playing the exact same playlist (50 tracks) over and over and it doesn't always happen on the same tracks either so I doubt you'll find the answer looking at specific tracks either

**Comment by murko69** [comment]
Could it be a cache-issue of sorts?
Don't know what 'cache spotify metadata' even does and that wasn't enabled. I just enabled it to test though.

**Comment by bluespark23** [comment]
In Snip I enabled "Spotify" , "Save Track History", "Display Track Pop-up", "Empty File If No Track Playing", with all other options disabled.  Also Updated WIndows to Windows 10 Professional (x64) Version 1909 (build 18363.720), Spotify updated to 1.1.28.721.g5b5ee660 and all appears to be OK again. In Windows notifications I enabled all options and disabled Focus Assist. All is working OK again since my last posting here. My knowledge of coding is minimal, but it occurred to me that the recent changes in Windows which included some .Net changes may have been responsible. From a user perspective it appears to be stable again. Thanks for looking into this, guys. Snip has on the whole has been rock solid for me so it's always external changes that seem to be the cause of any issues.

So for testing I have played a list of 30 titles -  of which were correctly tagged and titled files on my HDD, skipped back on forwards through the list with no errors. Will watch with interest to see if anyone finds a root cause.

Thanks again for an amazingly useful program!
